packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -475,7 +479,7 @@ LightningElement.prototype = {
             throw new Error('attachInternals API is not supported in this browser environment.');
         }
 
-        if (vm.renderMode === RenderMode.Light || vm.shadowMode === ShadowMode.Synthetic) {
+        if (vm.shadowMode === ShadowMode.Synthetic) {
```

Comment:
I discussed this internally with @nolanlawson and agreed to expose face callbacks to light DOM which means we need to expose `ElementInternals` to light DOM as well.